iam-role.tf

```
@@ -32,14 +32,14 @@ resource "aws_iam_role_policy_attachment" "cloudwatch_insights" {
 }
 
 resource "aws_iam_role_policy_attachment" "vpc_access" {
-  count = local.enabled && var.vpc_config == null ? 1 : 0
+  count = local.enabled && var.vpc_config != null ? 1 : 0
```

why not check the length instead ? This way it should work for empty values like `null` (default) and `{}` or a non-empty value.

```suggestion
  count = local.enabled && try(length(var.vpc_config), 0) > 0 ? 1 : 0
```